backend/src/api/routers/chains.py

```diff
@@ -0,0 +1,169 @@
+from fastapi import APIRouter
+from pydantic import BaseModel
+
+from ...chains.hybrid_retriever_chain import HybridRetrieverChain
+from ...chains.similarity_retriever_chain import SimilarityRetrieverChain
+from ...chains.multi_retriever_chain import MultiRetrieverChain
+
+from langchain_google_genai import ChatGoogleGenerativeAI
+
+from dotenv import load_dotenv
+
+
+class UserInput(BaseModel):
+    query: str
+    list_sources: bool = False
+    list_context: bool = False
+
+
+load_dotenv()
+
+llm = ChatGoogleGenerativeAI(model="gemini-pro", temperature=1)
+
+router = APIRouter(prefix="/chains", tags=["chains"])
+
+prompt_template_str = """
+    Use the following context:
+
+    {context}
+
+    -------------------------------------------------------------------------------------------------
+    Your task is to act as a knowledgeable assistant for users seeking information and guidance about the OpenROAD project. Avoid speculating or inventing information beyond the scope of the provided data.
+    Note that OR refers to OpenROAD and ORFS refers to OpenROAD-Flow-Scripts
+
+    Give a detailed answer to this question: 
+    {question}
+
+    """
+
+hybrid_retriever = HybridRetrieverChain(
+    llm_model=llm,
+    prompt_template_str=prompt_template_str,
+    embeddings_model_name="BAAI/bge-large-en-v1.5",
+    reranking_model_name="BAAI/bge-reranker-base",
```

Comment:
Have you tried also comparing the results with a smaller embedding/reranking models?

Reply:
Why is use_cuda True by default? Shouldn't it be a flag?

Reply:
I'm reading this from an env variable now. Thanks for pointing this out.